### PR TITLE
JSUI-2830 Make allowed values work with regex characters

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/JSUI-2830

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: src/controllers/FacetQueryController.ts [hunk 1]
```diff
@@ -379,7 +379,12 @@ export class FacetQueryController {
     // Replace the wildcard (*) for a regex match (.*)
     // Also replace the (?) with "any character once" since it is also supported by the index
     return _.some(this.facet.options.allowedValues, allowedValue => {
-      const regex = new RegExp(`^${allowedValue.replace(/\*/g, '.*').replace(/\?/g, '.')}$`, 'gi');
+      const regexContent = allowedValue
+        .replace(/[.+^${}()|[\]\\]/g, '\\$&')
+        .replace(/\*/g, '.*')
+        .replace(/\?/g, '.');
```
Comment (btaillon) [inline src/controllers/FacetQueryController.ts]: Would it be a good idea to split this across multiple variables and/or multiple functions for readability?

Comment (ThibodeauJF) [reply src/controllers/FacetQueryController.ts]: you don't think having this on multiple lines is readable enough?

File: src/controllers/FacetQueryController.ts [hunk 2]
```diff
@@ -379,7 +379,12 @@ export class FacetQueryController {
     // Replace the wildcard (*) for a regex match (.*)
     // Also replace the (?) with "any character once" since it is also supported by the index
     return _.some(this.facet.options.allowedValues, allowedValue => {
-      const regex = new RegExp(`^${allowedValue.replace(/\*/g, '.*').replace(/\?/g, '.')}$`, 'gi');
+      const regexContent = allowedValue
+        .replace(/[.+^${}()|[\]\\]/g, '\\$&')
```
[the remark this answers is not in the document]
Comment (ThibodeauJF) [reply src/controllers/FacetQueryController.ts]: >  Have you seen this stackoverflow post?
I have taken it from the Mozilla JS Regular Expression doc https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions Escaping section and I removed the * and ^ from it already.

I will move it out of the method to make it clearer no problem.

[the remark this answers is not in the document]
Comment (ThibodeauJF) [reply src/controllers/FacetQueryController.ts]: Looks like it work with dashes and forward slashes without needing to add them, I'll keep the regexp from the doc.

**Pull request merged**